src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,888 @@
+      G.descr = Symbol("Omega+")
+   elseif e==-1
+      iseven(n) || throw(ArgumentError("The dimension must be even"))
+      G = MatrixGroup(n,GF(b,a)[1])
+      G.descr = Symbol("Omega-")
+   elseif e==0
+      isodd(n) || throw(ArgumentError("The dimension must be odd"))
+      G = MatrixGroup(n,GF(b,a)[1])
+      G.descr = :Omega
+   else
+      throw(ArgumentError("Invalid description of orthogonal group"))
+   end
+   return G
+end
+
+omega_group(n::Int, q::Int) = omega_group(0,n,q)
+
+"""
+    unitary_group(n::Int, q::Int)
+    GU = unitary_group
+
+Return the unitary group of dimension `n` over the field `GF(q^2)`.
+"""
+function unitary_group(n::Int, q::Int)
+   (a,b) = ispower(q)
+   if !isprime(b) throw(ArgumentError("The field size must be a prime power")) end
+   G = MatrixGroup(n,GF(b,2*a)[1])
+   G.descr = :GU
+   return G
+end
+
+function unitary_group(::Type{T}, n::Int, q::Int) where T <: GAPGroup
+   if T<:PermGroup   return T(GAP.Globals.GU(_gap_filter(T), n, q))
+   elseif T<:MatrixGroup   return unitary_group(n,q)
+   else throw(ArgumentError("Type not supported"))
+   end
+end
+
+"""
+    special_unitary_group(n::Int, q::Int)
+    SU = special_unitary_group
+
+Return the special unitary group of dimension `n` over the field `GF(q^2)`.
+"""
+function special_unitary_group(n::Int, q::Int)
+   (a,b) = ispower(q)
+   if !isprime(b) throw(ArgumentError("The field size must be a prime power")) end
+   G = MatrixGroup(n,GF(b,2*a)[1])
+   G.descr = :SU
+   return G
+end
+
+function special_unitary_group(::Type{T}, n::Int, q::Int) where T <: GAPGroup
+   if T<:PermGroup   return T(GAP.Globals.SU(_gap_filter(T), n, q))
+   elseif T<:MatrixGroup   return special_unitary_group(n,q)
+   else throw(ArgumentError("Type not supported"))
+   end
+end
+
+const GL = general_linear_group
+const SL = special_linear_group
+const Sp = symplectic_group
+const GO = orthogonal_group
+const SO = special_orthogonal_group
+const GU = unitary_group
+const SU = special_unitary_group
+
+
+"""
+    matrix_group(V::T...) where T<:MatrixGroup
+    matrix_group(V::AbstractVector{T}) where T<:MatrixGroup
+
+Return the matrix group generated by elements in the vector `V`.
+"""
+function matrix_group(V::AbstractVector{T}) where T<:Union{MatElem,MatrixGroupElem}
+   if T<:MatElem      # if T <: MatrixGroupElem, we can already assume that det(V[i]) != 0 
+      for v in V @assert det(v) !=0 "The matrix is not invertible" end
+   end
+   return MatrixGroup(nrows(V[1]),base_ring(V[1]),V)
+end
+
+matrix_group(V::T...) where T<:Union{MatElem,MatrixGroupElem} = matrix_group(collect(V))
+
+
+
+
+########################################################################
+#
+# Subgroups
+#
+########################################################################
+
+function sub(G::MatrixGroup, elements::Vector{S}) where S <: GAPGroupElem
+   @assert elem_type(G) == S
+   elems_in_GAP = GAP.julia_to_gap(GapObj[x.X for x in elements])
+   H = GAP.Globals.Group(elems_in_GAP)
```

Comment:
Probably better to have this tracked as a subgroup in GAP, too (so it can use knowledge from the parent group)... 
```suggestion
   H = GAP.Globals.Subgroup(G.X, elems_in_GAP)
```
The same should be fixed in the general `sub` for GAPGroup.

Except that perhaps `G.X` might not be defined?!? Perhaps need a separate case for that?

Reply:
For the moment, we decided that group homomorphisms always have an underlying GAP object, and that the methods `sub(G, [...])` always returns the subgroup `H` together with the inclusion homomorphism of `H` into `G`. Hence, every time we compute the subgroup of a matrix group `G`, the underlying GAP object is always computed as well.